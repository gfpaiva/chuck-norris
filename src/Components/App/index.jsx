import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import Spinner from 'react-spinkit';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Route, Switch } from 'react-router-dom';

import CategoriesActions from '../../Store/categories/actions';

import Footer from '../Footer';
import Routes from '../../Utils/Routes';

import './App.scss';

function App({ loading, loadCategoriesRequest }) {
  useEffect(() => {
    loadCategoriesRequest();
  }, []);

  if (loading) {
    return (
      <div className="app__loading full full--centered">
        <Spinner name="ball-pulse-sync" />
      </div>
    );
  }

  return (
    <main className="app">
      <section className="app__content container py-3">
        <Switch>
          {Routes.map(route => (
            <Route {...route} />
          ))}
        </Switch>
      </section>
      <Footer />
    </main>
  );
}

App.propTypes = {
  loading: PropTypes.bool.isRequired,
  loadCategoriesRequest: PropTypes.func.isRequired,
};

const mapStateToProps = ({ categories }) => ({ loading: categories.loading });
const mapDispatchToProps = dispatch => bindActionCreators(CategoriesActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(App);
