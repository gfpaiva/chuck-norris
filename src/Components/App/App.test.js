import React from 'react';
import wait from 'waait';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import store from '../../Store';

import App from './index';

import getCategories from '../../../__mocks__/getCategories';

describe('<App />', () => {
  it('should mount properly', async () => {
    fetch
      .once(JSON.stringify(getCategories));

    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    // Before get fetch data (loading state)
    expect(wrapper.find('App')).toMatchSnapshot();

    await wait();
    wrapper.update();

    // Full load
    expect(wrapper.find('App')).toMatchSnapshot();
  })
});
