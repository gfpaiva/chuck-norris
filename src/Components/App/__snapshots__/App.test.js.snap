// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should mount properly 1`] = `
<App
  loadCategoriesFail={[Function]}
  loadCategoriesRequest={[Function]}
  loadCategoriesSuccess={[Function]}
  loading={true}
>
  <div
    className="app__loading full full--centered"
  >
    <Spinner
      fadeIn="full"
      name="ball-pulse-sync"
      noFadeIn={false}
      overrideSpinnerClassName=""
    >
      <div
        className="sk-fade-in sk-spinner ball-pulse-sync"
      >
        <div
          key="0"
        />
        <div
          key="1"
        />
        <div
          key="2"
        />
      </div>
    </Spinner>
  </div>
</App>
`;

exports[`<App /> should mount properly 2`] = `
<App
  loadCategoriesFail={[Function]}
  loadCategoriesRequest={[Function]}
  loadCategoriesSuccess={[Function]}
  loading={false}
>
  <main
    className="app"
  >
    <section
      className="app__content container py-3"
    >
      <Switch>
        <Route
          component={[Function]}
          computedMatch={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          exact={true}
          key="home"
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          path="/"
        >
          <Home
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="home"
            >
              <div
                className="home__welcome text-center"
              >
                <img
                  alt="Chuck Norris"
                  className="home__welcome-image full-width"
                  src="test-file-stub"
                  title="Chuck Norris"
                />
                <h1
                  className="home__welcome-title bg-secondary py-1"
                >
                  <span>
                    Chuck Facts
                  </span>
                </h1>
                <p
                  className="home__welcome-text"
                >
                  Selecione uma das categorias abaixo para "Chuck Facts"
                </p>
              </div>
              <Connect(CategoryGrid)>
                <CategoryGrid
                  active=""
                  categories={
                    Array [
                      "animal",
                      "career",
                      "celebrity",
                      "dev",
                      "explicit",
                      "fashion",
                      "food",
                      "history",
                      "money",
                      "movie",
                      "music",
                      "political",
                      "religion",
                      "science",
                      "sport",
                      "travel",
                    ]
                  }
                  dispatch={[Function]}
                >
                  <div
                    className="category-grid"
                  >
                    <Grid
                      items={Array []}
                    >
                      <div
                        className="grid"
                      >
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/animal"
                          key="ANIMAL"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/animal"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/animal"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üêµ
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                ANIMAL
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/career"
                          key="CAREER"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/career"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/career"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üë®‚Äçüíº
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                CAREER
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/celebrity"
                          key="CELEBRITY"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/celebrity"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/celebrity"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üë®‚Äçüé®
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                CELEBRITY
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/dev"
                          key="DEV"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/dev"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/dev"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                ü§ì
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                DEV
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/explicit"
                          key="EXPLICIT"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/explicit"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/explicit"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üîû
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                EXPLICIT
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/fashion"
                          key="FASHION"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/fashion"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/fashion"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üëó
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                FASHION
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/food"
                          key="FOOD"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/food"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/food"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üçï
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                FOOD
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/history"
                          key="HISTORY"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/history"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/history"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üìú
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                HISTORY
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/money"
                          key="MONEY"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/money"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/money"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üí∏
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                MONEY
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/movie"
                          key="MOVIE"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/movie"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/movie"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üé•
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                MOVIE
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/music"
                          key="MUSIC"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/music"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/music"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üéµ
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                MUSIC
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/political"
                          key="POLITICAL"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/political"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/political"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                ‚öñÔ∏è
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                POLITICAL
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/religion"
                          key="RELIGION"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/religion"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/religion"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                ‚õ™
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                RELIGION
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/science"
                          key="SCIENCE"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/science"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/science"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üî¨
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                SCIENCE
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/sport"
                          key="SPORT"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/sport"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/sport"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üö¥
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                SPORT
                              </span>
                            </a>
                          </Link>
                        </Card>
                        <Card
                          className="category-grid__item text-center px-0 py-4"
                          hover={true}
                          href="/category/travel"
                          key="TRAVEL"
                        >
                          <Link
                            className="card card--hover category-grid__item text-center px-0 py-4"
                            to="/category/travel"
                          >
                            <a
                              className="card card--hover category-grid__item text-center px-0 py-4"
                              href="/category/travel"
                              onClick={[Function]}
                            >
                              <span
                                className="category-grid__icon"
                              >
                                üß≥
                              </span>
                              <br />
                              <span
                                className="category-grid__name"
                              >
                                TRAVEL
                              </span>
                            </a>
                          </Link>
                        </Card>
                      </div>
                    </Grid>
                  </div>
                </CategoryGrid>
              </Connect(CategoryGrid)>
            </div>
          </Home>
        </Route>
      </Switch>
    </section>
    <Memo(Footer)>
      <footer
        className="footer text-center"
      >
        <p>
          Feito com
          <span
            aria-label="Cora√ß√£o"
            role="img"
          >
            Ô∏è ‚ù§Ô∏è 
          </span>
          por:¬†
          <a
            className="linked"
            href="https://guilherme.dev/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Guilherme Paiva
          </a>
        </p>
      </footer>
    </Memo(Footer)>
  </main>
</App>
`;
